src/bosh-director/lib/bosh/director/agent_broadcaster.rb

```diff
@@ -79,24 +83,25 @@ def sync_dns(instances, blobstore_id, sha1, version)
           pending_clone = pending.clone
         end
 
+        start_time_cancelling = Time.now
         unresponsive_agents = []
         pending_clone.each do |instance|
-          agent_client = agent_client(instance.agent_id, instance.name)
-          agent_client.cancel_sync_dns(instance_to_request_id[instance])
+          @nats_rpc.cancel_request(instance_to_request_id[instance])
```

Comment:
replacing `cancel_sync_dns` with direct rpc call kind of confusing as it don't explain what we are actually canceling until you look around the code. For maintainability aspect the code it look more cleaner approach, but kind of hiding the intent as the rpc method is very generic (as the agent method is just a wrapper around the rpc call). Just a view...

Reply:
Good catch, this seems to have gone wrong during squashing the changes.